### PR TITLE
flake: nixosTests refactor

### PR DESCRIPTION
**Pull request by mightyiam**
Does it make it more readable? Not sure. Maybe. How do you feel about it?

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -29,21 +29,25 @@
...
+                mkNetwork = import "${dir}/${name}";
+                network = mkNetwork {
                   inherit pkgs;
                   inherit (pkgs) lib;
                   modules = extendedModules;
                   configurations = importNixosConfigurations;
-                })) (readDir dir)
+                };
+              in
+                pkgs.nixosTest network;
```
Comment (lorenzleutgeb) [inline flake.nix:48]: I am not sure that the argument passed to `nixosTest` should be called a "network". It does contain `nodes` that will end up networked, but it also contains the test script itself, and some other metadata (test name) and configuration (for interactive testing). The [NixOS manual says]: "We refer to the whole test above as a test module, whereas the values in [`nodes.<name>`](https://nixos.org/manual/nixos/stable/#test-opt-nodes) are NixOS modules themselves."

So, I'd go for `testModule` instead.
```suggestion
                pkgs.nixosTest testModule;
```

**Review by lorenzleutgeb — CHANGES_REQUESTED** [review]

**Comment by lorenzleutgeb** [comment]
Does make it more readable. However, not sure about the name "network".

**Comment by mightyiam** [comment]
The name network I picked up from the nixpkgs manual. I found this function and the parameter is named network. Sorry, I'm on phone. @lorenzleutgeb .

File: flake.nix (hunk 1 above)
Comment (mightyiam) [reply flake.nix:48]: I agree.

**Pull request merged**